x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -15,6 +15,10 @@ import (
 
 // Function for reputers to call to add stake to an existing stake position.
 func (ms msgServer) AddStake(ctx context.Context, msg *types.MsgAddStake) (*types.MsgAddStakeResponse, error) {
+	if err := msg.Validate(); err != nil {
```

Comment:
did you do a full sweep of the msgserver? seems like we should also have this in multiple places even within each of the add/remove delegate functions and elsewhere